### PR TITLE
14 create a search component AOA

### PR DESCRIPTION
**Pull request by AOA19**
## What changes did I make?
- Created the SearchBar component based on the Figma design and passed in the props from the API request in App.jsx
- Deleted the temporary search bar that was inside the App.jsx component 
- Removed styling of the temporary search bar in index.css and replaced it with spacing reset for the whole app
- Added a css file for the SearchBar component 
- Added the react-icons library as a dependency
- Deleted boilerplate styling from App.css
- Deleted .eslintcache file and added to gitignore file 
- Removed "jsx" extension for App import inside of main.jsx

## Why did you make these changes?
- Needed to create a separate search bar component for the app 
- Index.css will be used for global styling only that's why I reset the spacing of the entire app. Any styling for the App.jsx component will go inside of App.css

## Visual changes (if applicable)
- Empty search bar 
search-bar-1

- Search bar with in
search-bar-2
put

### CODE REVIEW

**Comment by allison-truhlar** [comment]
@AOA19 This search component is looking good! I have one edit to make and a couple of suggestions/questions:

To edit:
- I think we need to also have ingredients sent as a prop to the search form, and then inside the form, add a property, value={ingredients}, in order to make the form a controlled input.

Suggestions/questions (do not need to be addressed before merging):
- I think we will need to add some sort of instructions/validation to the user input to ensure each ingredient is separated by a space for searching. We could also consider changing to a separate input per ingredient, as discussed on Discord.
- I see you have an onReset function in the form - I'm not familiar with this. After some Googling, I'm wondering - do we need to add a reset button for this function to work? If not, when is it called?

**Comment by AOA19** [comment]
@allison-truhlar, @Emimint, @komalgill0310 
- Instead of passing ingredients as a prop to the search bar I moved the whole useState hook to SearchBar.jsx. It made more sense to have it in the SearchBar component because it was the only component that was using it. So I only pass the fetchData function as a prop to the SearchBar component. 
- I left the recipes and error hooks in App.jsx since other components will be using those hooks. I commented out the isSearching hook because I wasn't sure if we needed it anymore or not.
- I also move the handleSearch function from App.jsx to SearchBar.jsx. It made more sense to write that function inside the search component. And inside that function I added the validation that you suggested.
- An error will display when the search bar is submitted without an input (*see image 1*) and an error will display when the input is an invalid format (*see image 2*), based on the regular expression. 
- The valid format is any letter (not case sensitive), at least 3 characters long, no numbers, required space separating each word and an optional comma. 


**Image 1**
empty-input-error


**Image 2**
invalid-format-error

File: src/App.jsx [hunk 1]
```diff
@@ -37,56 +38,27 @@ function App() {
       if (response.data.count > 0) {
```
Comment (robester0403) [inline src/App.jsx:41]: You don't need an if statement here. Unless you have you have a specific message to display if it comes back with 0 results.

File: src/App.jsx [hunk 2]
```diff
@@ -5,18 +5,19 @@ import axios from 'axios';
...
   const [recipes, setRecipes] = useState(null);
-  const [isSearching, setIsSearching] = useState(false);
+  // const [isSearching, setIsSearching] = useState(false);
   const [error, setError] = useState('Add an ingredient to search!');
 
   // This function could potentially be moved out to a "utils" folder?
-  // In that case, I think the response or error could be returned, and then used with setRecipes or setError here
+  // In that case, I think the response or error could be returned,
+  // and then used with setRecipes or setError here
   async function fetchData(ingredientString) {
```
Comment (robester0403) [inline src/App.jsx:21]: This is an api fetching function and should be refactored from this layer. This file does not specifically focus on this layer of logic. It should be exported from an API file.

Comment (robester0403) [inline src/App.jsx:21]: In addition, you need to reset the error message so the new search clears the error messaging from the previous search.

File: src/App.jsx [hunk 3]
```diff
@@ -5,18 +5,19 @@ import axios from 'axios';
 
 import RecipeBrief from './components/RecipesDisplay/RecipeBrief';
 import Header from './components/header/Header';
+import SearchBar from './components/searchBar/SearchBar';
 
 function App() {
```
Comment (robester0403) [inline src/App.jsx:12]: You should reserve the App file for more providers and context related wrappers. And wrap them around a main page file that should hold all the logic you have currently on this page.

**Review by robester0403 — COMMENTED** [review]
Initial comments

**Review by allison-truhlar — APPROVED** [review]

**Pull request merged**